### DISCUSSION TITLE
주문 처리 상태 변화

### DISCUSSION THREAD

**Discussion by Bue-von-hon** [f-lab-edu/foody-lab#10] [Ideas]
주문은 생성 이후 아래와 같은 과정을 거칩니다.
주문 생성 -> 결제 완료 -> 마일리지 적립 요청

이 과정에서 주문 생성이 실패하거나, 결제가 실패하면 모든 과정이 취소되어야 합니다.
마일리지 적립은 실패하여도 해당 부분만 재시도하면 됩니다.

**Comment by Taewoongjung** [comment]
"마일리지 적립 요청 이벤트 생성"은 어떻게 생각하시나요?

**Reply by Bue-von-hon** [reply]
별개라고 생각해요
마일리지 적립은 실패하면 해당 부분만 재시도하면 된다 생각합니다
***결제가 실패하면 주문 생성도 취소되어야할지 의견을 묻고 싶어요!***

**Reply by Taewoongjung** [reply]
이벤트 소싱처럼 이벤트를 저장해놓는게 생각이 나서 이벤트라고 생각이 되어졌네요. 마일리지는 매일 12시마다 처리되는거니까요

**Reply by Bue-von-hon** [reply]
주문과 결제는 어떻게 생각하시나요?
마일리지는 같은 트랜잭션에 포함되지 **않아도** 된다고 생각합니다!

**Reply by Taewoongjung** [reply]
네 저도 동의해요!

**Comment by Bue-von-hon** [comment]
조금 더 생각을 해보니 주문 생성과 결제는 아예 개별적인 것 같습니다!
